packages/labs/task/src/task.ts

```diff
@@ -215,9 +221,11 @@ export class Task<
         if (error === undefined) {
           this.status = TaskStatus.COMPLETE;
           this._resolveTaskComplete(result as R);
+          this._onComplete?.(result as R);
```

Comment:
I think this needs to be in a try/catch block to keep from messing with task state if it throws. Same for onError.

Reply:
what should we do on catch? It's already too late to report onError, no?

Reply:
Something like this?

```ts
if (error === undefined) {
  try {
    this._onComplete?.(result as R);
  } catch {
    // Ignore user errors from onComplete.
  }
  this.status = TaskStatus.COMPLETE;
  this._resolveTaskComplete(result as R);
} else {
  try {
    this._onError?.(error);
  } catch {
    // Ignore user errors from onError.
  }
  this.status = TaskStatus.ERROR;
  this._rejectTaskComplete(error);
}
this._value = result as R;
this._error = error;
 ```

Reply:
exactly